geode-core/src/main/java/org/apache/geode/cache/execute/Execution.java

```diff
@@ -106,6 +107,22 @@
    */
   ResultCollector<OUT, AGG> execute(String functionId) throws FunctionException;
 
+  /**
+   * Executes the function using its {@linkplain Function#getId() id} with the specified timeout.
+   * <p>
+   * {@link Function#execute(FunctionContext)} is called on the instance retrieved using
+   * {@link FunctionService#getFunction(String)} on the executing member.
+   *
+   * @param functionId id of the function to execute
+   * @param timeoutMs milliseconds to wait for the operation to finish before timing out
+   * @return ResultCollector to retrieve the results received. This is different object than the
+   *         ResultCollector provided in {@link Execution#withCollector(ResultCollector)}. User has
+   *         to use this reference to retrieve results.
+   *
+   * @since GemFire 6.0
+   */
+  ResultCollector<OUT, AGG> execute(String functionId, int timeoutMs) throws FunctionException;
```

Comment:
There was a discussion months ago about migrating timeout params to `long` and `TimeUnit` pairs on all public APIs. Please use signature:
`ResultCollector<OUT, AGG> execute(String functionId, long timeout, TimeUnit unit) throws FunctionException;`

Reply:
Here is the discussion https://github.com/apache/geode/pull/3530

Reply:
I will. Thanks.
By the way, the timeout passed to the execute and executeFunction methods of DistributedRegionFunctionExecutor, PartitionedRegionFunctionExecutor, MemberFunctionExecutor and MultiRegionFunctionExecutor will have no effect.
Is that ok? If so, should I highlight it in the code somehow?